SmartDeviceLink/SDLHapticManager.m

```diff
@@ -0,0 +1,150 @@
+//
+//  SDLHapticManager.m
+//  SmartDeviceLink-iOS
+//
+//  Copyright © 2017 smartdevicelink. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "SDLHapticManager.h"
+#import "SDLNotificationConstants.h"
+#import "SDLRectangle.h"
+#import "SDLHapticRect.h"
+#import "SDLSendHapticData.h"
+#import "SDLTouch.h"
+
+@interface SDLHapticManager()
+
+/*!
+ *  @abstract
+ *      The projection window associated with the Haptic Manager
+ */
+@property (nonatomic, strong) UIWindow *projectionWindow;
+
+/*!
+ *  @abstract
+ *      Array of focusable view objects extracted from the projection window
+ */
+@property (nonatomic, strong) NSMutableArray<UIView *> *focusableViews;
+
+/*!
+ *  @abstract
+ *      reference to SDLManager
+ */
+@property (nonatomic, strong) SDLManager *sdlManager;
+@end
+
+
+@implementation SDLHapticManager
+
+- (instancetype)initWithWindow:(UIWindow *)window sdlManager:(SDLManager *)sdlManager{
+    if ((self = [super init])) {
+        self.projectionWindow = window;
+        self.sdlManager = sdlManager;
+        [self updateInterfaceLayout];
+        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(projectionViewUpdated:) name:SDLProjectionViewUpdate object:nil];
+        
+    }
+    return self;
+}
+
+- (void)updateInterfaceLayout {
+    self.focusableViews = [NSMutableArray new];
+    [self parseViewHierarchy:[[self.projectionWindow subviews] lastObject]];
+    
+    NSUInteger preferredViewIndex = [self.focusableViews indexOfObject:[[[self.projectionWindow subviews] lastObject] preferredFocusedView]];
```

Comment:
Correct, this will not support a `UIButton` if that button type does not support 'focusable' testing via the `canBecomeFocused` property.
In order to use a `UIButton` with the haptic manager, it will somehow need to be flagged as `canBecomeFocused`.
As long as a `UIButton` or any other special view is flagged properly, the `parseViewHierarchy` function should work as expected.
It seems like it might be cleaner to handle the special case logic for flagging a button outside of the haptic manager itself.